### ISSUE TITLE
azure_rm_virtualmachine using "latest" as image version does not select actual latest version

### ISSUE THREAD

**Issue by Klaas-** [ansible/ansible_collections_azure#1667]
##### SUMMARY
using "latest" as the image version does not select the actual latest. I initially thought "latest" is send to API directly (because API does accept that) but it seems there is some magic going on there in the module.

At the end I think the problem is from here:
https://github.com/ansible-collections/azure/blob/997e4de875fa12e99c9a064b614797a65c58677e/plugins/modules/azure_rm_virtualmachine.py#L2780-L2786
https://learn.microsoft.com/en-us/python/api/azure-mgmt-compute/azure.mgmt.compute.v2024_03_01.operations.virtualmachineimagesoperations?view=azure-python#azure-mgmt-compute-v2024-03-01-operations-virtualmachineimagesoperations-list

where ordering by name is lexicographical instead of numerical

Having said that -- I think the correct answer here is just sending "latest" to the API

References https://github.com/Azure/azure-rest-api-specs/issues/30142

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure_rm_virtualmachine

##### ANSIBLE VERSION

```paste below
2.15
```

##### COLLECTION VERSION

```paste below
2.6.0
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

RHEL9

##### STEPS TO REPRODUCE

```yaml
- name: Create VM in Azure
  azure.azcollection.azure_rm_virtualmachine:
    subscription_id: "{{ subscription_id }}"
    name: "testvm"
    resource_group: "{{ resource_group_name }}"
    vm_size: "{{ vm_size }}"
    network_interface_names: "{{ nic_id }}"
    os_disk_name: "testvm-disk1-os"
    created_nsg: false
    os_type: Linux
    image:
      publisher: RedHat
      offer: RHEL
      sku: 8-lvm-gen2
      version: latest
    ssh_password_enabled: false
    admin_username: username
    ssh_public_keys:
      - path: /home/username/.ssh/authorized_keys
        key_data: "{{ ssh_public_key }}"
    managed_disk_type: Premium_LRS
    boot_diagnostics:
      enabled: true
      type: managed
    security_profile:
      encryption_at_host: true
      security_type: TrustedLaunch
      uefi_settings:
        secure_boot_enabled: true
        v_tpm_enabled: true
    state: present
    started: true
    vm_identity:
      type: SystemAssigned
```

##### EXPECTED RESULTS

RHEL 8.10 image

##### ACTUAL RESULTS

RHEL 8.9 image

**Comment by Fred-sun** [comment]
@Klaas- At present, the API version is only released to v2024-03-02, and the method of obtaining the latest image version is also list. Do you have any better suggestions?

**Comment by Fred-sun** [comment]
@Klaas- In addition, I can confirm that sorting by 'name' returns an ordered list of image versions, and that the last value in this list is the latest image version. Thank you!

**Comment by Fred-sun** [comment]
Fixed by #1669
